common/network-common/src/main/java/org/apache/spark/network/util/ByteArrayReadableChannel.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.util;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+
+import io.netty.buffer.ByteBuf;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ByteArrayReadableChannel implements ReadableByteChannel {
+  private final Logger logger = LoggerFactory.getLogger(ByteArrayReadableChannel.class);
+
+  private byte[] data;
+  private int offset;
+
+  public byte[] getData() {
+    return data;
+  }
+
+  public int length() {
+    return data.length;
+  }
+
+  public void reset() {
+    offset = 0;
+    data = null;
+  }
+
+  public ByteArrayReadableChannel() {
+
+  }
+
+  public void feedData(ByteBuf buf) {
+    int length = buf.readableBytes();
+
+    if (buf.hasArray()) {
+      data = buf.array();
+      buf.skipBytes(length);
+    } else {
+      data = new byte[length];
```

Comment:
Do you need to do this using byte arrays? Why not use `ByteBuffer.put(ByteBuffer)`?

Also this code looks very suspicious. You're overwriting `data` but you don't reset `offset`. Which tells me that you're trying to handle short writes and doing it incorrectly.